fix get_class($this) in relationship functions

bebnev opened the pull request:
The request fixes the issue #239 

The problem was in the strpos() in exploading argument list. It was looking for the first occurance of '(' which is in get_class($this), so it takes everything before ')' --> get_class($this. While in getClassName function strpos is checking - strpos($className, 'get_class($this)') !== false, so it fails. 
I have changed it to strrpos - in order to find the last occurance and explode the full argument list.

barryvdh commented:
Hmm, not sure if this fixes all the edge cases. Should probably need some test or make a better parser. What if there are chained methods calls? `$this->belongsTo(get_class($this))->orderBy('id')`?
Can't you use `static::class`? Or we could check for `get_class($this` instead.

bebnev commented:
Yee, I have totally forgotten about chain methods. You are right. 
So the simpliest workaround in that case would be - a check for `get_class($this`

barryvdh commented:
Can you verify that that works and submit a new PR? (Or fix this PR and squash?)
Thanks!

bebnev commented:
Yes, I'll do it today.

bebnev commented:
Hi. When you have time, please have a look. I've changed a bit the way of parsing relations function. I've taken into consideration different preferences of writing chaining functions, so I've make one string from the function code and parse it. Also I've changed the way of parsing arguments.
I've checked this on several different tests:

```php
public function posts()
    {
        return $this
            ->hasMany('App\Models\Post')
            ->orderBy('date_to')
            ->limit(10);
    }

public function phones()
    {
        return $this->hasOne('App\Phone');
    }

public function posts2()
    {
        return $this
            ->belongsToMany(static::class, 'table1', 'id4', 'id5')->orderBy('date');
    }

public function posts3()
    {
        return $this
            ->hasMany(Post::class)->orderBy('date_to')
            ->limit(10);
    }

    public function posts4()
    {
        return $this->hasMany(get_class($this));
    }
```
And as a result I have:
```
@property-read \Illuminate\Database\Eloquent\Collection|\App\Models\Post[] $posts
@property-read \App\Phone $phones
@property-read \Illuminate\Database\Eloquent\Collection|\App\User[] $posts2
@property-read \Illuminate\Database\Eloquent\Collection|Post[] $posts3
@property-read \Illuminate\Database\Eloquent\Collection|\App\User[] $posts4 
```

barryvdh commented:
Thanks!

Adelf commented:
Hello. Seems this patch broke something.

    public function roles()
    {
        return $this->belongsToMany(Config::get('entrust.role'), Config::get('entrust.role_user_table'), 'user_id', 'role_id');
    }

Now creates:

    * @property-read \Illuminate\Database\Eloquent\Collection|\Config::get('entrust.role[] $roles 

Previous version created: 

    * @property-read \Illuminate\Database\Eloquent\Collection|\Config::get('entrust.role')[] $roles 

which was a bit better :)

bebnev commented:
@Adelf Hello. Which version of the ide-helper are you using ? 
I have just installed 2.1.* and the generation is okay:

I have 
```php
public function roles()
    {
        return $this->belongsToMany(Config::get('entrust.role'), Config::get('entrust.role_user_table'), 'user_id', 'role_id');
    }
```
in the beginning, and 
```php
* @property-read \Illuminate\Database\Eloquent\Collection|\Config::get('entrust.role')[] $roles
```
after generation

But in version 2.0.6 I have the problem you have described.

Adelf commented:
@bebnev sorry. You're right. My fault.
